platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/ConsensusEngine.java

```diff
@@ -16,139 +16,24 @@
-    }
-
+public interface ConsensusEngine {
     /**
      * Add an event to the hashgraph
      *
      * @param event an event to be added
      * @return a list of rounds that came to consensus as a result of adding the event
      */
     @NonNull
-    public List<ConsensusRound> addEvent(@NonNull final EventImpl event) {
-        Objects.requireNonNull(event);
-
-        try {
-            if (event.getGeneration() < consensusSupplier.get().getMinGenerationNonAncient()) {
-                // ancient events *may* be discarded, and stale events *must* be discarded
-                return List.of();
-            }
-
-            final long minimumGenerationNonAncientBeforeAdding =
-                    consensusSupplier.get().getMinGenerationNonAncient();
-
-            // record the event in the hashgraph, which results in the events in consEvent reaching consensus
-            final List<ConsensusRound> consensusRounds = consensusSupplier.get().addEvent(event);
-
-            eventAddedMetrics.eventAdded(event);
-
-            final long minimumGenerationNonAncient = consensusSupplier.get().getMinGenerationNonAncient();
-
-            if (minimumGenerationNonAncient > minimumGenerationNonAncientBeforeAdding) {
-                // consensus rounds can be null and the minNonAncient might change, this is probably because of a round
-                // with no consensus events, so we check the diff in generations to look for stale events
-                handleStale(minimumGenerationNonAncientBeforeAdding);
-            }
-
-            return Objects.requireNonNullElseGet(consensusRounds, List::of);
-        } finally {
-            intakeEventCounter.eventExitedIntakePipeline(event.getBaseEvent().getSenderId());
-        }
-    }
-
-    /**
-     * Notify observer of stale events, of all event in the consensus stale event queue.
-     *
-     * @param previousGenerationNonAncient the previous minimum generation of non-ancient events
-     */
-    private void handleStale(final long previousGenerationNonAncient) {
-        // find all events that just became ancient and did not reach consensus, these events will be considered stale
-        final Collection<EventImpl> staleEvents = shadowGraph.findByAncientIndicator(
-                previousGenerationNonAncient,
-                consensusSupplier.get().getMinGenerationNonAncient(),
-                ConsensusEngine::isNotConsensus);
-
-        for (final EventImpl staleEvent : staleEvents) {
-            staleEvent.setStale(true);
-            staleMetrics.staleEvent(staleEvent);
-            staleEventConsumer.accept(staleEvent);
-        }
-    }
-
-    /**
-     * Returns true if the event has not reached consensus
-     *
-     * @param event the event to check
-     * @return true if the event has not reached consensus
-     */
-    private static boolean isNotConsensus(@NonNull final EventImpl event) {
-        return !event.isConsensus();
-    }
+    @InputWireLabel("event")
```

Comment:
If the label was `EventImpl` it would be clear on the diagram that its not a `GossipEvent` that goes across the wire. (Thinking out loud)

Reply:
I think that's a good change. I'm liking the class-name-as-wire-label pattern. Will push a change